transport/grpc/stream.go

```diff
@@ -133,6 +148,26 @@ func (cs *clientStream) Close(context.Context) error {
 	return cs.stream.CloseSend()
 }
 
+func (cs *clientStream) Headers() (transport.Headers, error) {
+	md, err := cs.stream.Header()
+	if err != nil {
+		return transport.NewHeaders(), err
+	}
+	headers := transport.NewHeadersWithCapacity(len(md))
+	for k, vs := range md {
+		v := vs[0]
```

Comment:
I think we also need to check for the case when there's a key but no values.
```go
var v string
if len(vs) > 1 {
 v = vs[0]
}

Reply:
Yeah, noticed that added a `if len(vs) != 1` check.